lib/config/validation.spec.ts

```diff
@@ -920,5 +921,56 @@ describe('config/validation', () => {
         },
       ]);
     });
+
+    it('errors if unallowed header in hostRules', async () => {
+      GlobalConfig.set({ allowedHeader: ['X-*'] });
+
+      const config = {
+        hostRules: [
+          {
+            matchHost: 'https://domain.com/all-versions',
+            header: {
+              'X-Auth-Token': 'token',
+              unallowedHeader: 'token',
+            },
+          },
+        ],
+      };
+      const { warnings, errors } =
+        await configValidation.validateConfig(config);
+      expect(warnings).toHaveLength(0);
+      expect(errors).toMatchObject([
+        {
+          message:
+            "hostRules header `unallowedHeader` is not permitted by this bot's `allowedHeader`.",
```

Comment:
I don't like the word `unallowedHeader`. Are Renovate users going to see that message? If yes, can we use a verb like `forbidden` instead?

Reply:
The error message is dynamic, `unallowedHeader` appears because in the test I set an field `unallowedHeader` inside header. Users will not see this message